### PR TITLE
Fixed decal automatic technique editor selection to match runtime

### PR DESCRIPTION
**Pull request by lukaschod**
### Purpose of this PR
Refactoring decal automatic selection to work more closely like editor one.
Continue of https://github.com/Unity-Technologies/Graphics/pull/6731.
Also fixes https://fogbugz.unity3d.com/f/cases/1370326/.

---
### Testing status with automatic technique with decals
- [x] Build WebGL
- [x] Build WindowsPlayer/WindowsEditor
- [ ] Build Console
- [x] Build Mobile

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs [hunk 1]
```diff
@@ -313,24 +313,52 @@ private DecalTechnique GetTechnique(bool isDeferred)
...
+                // Console
+                case UnityEditor.BuildTargetGroup.GameCoreXboxOne:
+                case UnityEditor.BuildTargetGroup.GameCoreXboxSeries:
+                case UnityEditor.BuildTargetGroup.PS4:
+                case UnityEditor.BuildTargetGroup.PS5:
+                case UnityEditor.BuildTargetGroup.Switch:
+                    return true;
+
+                default:
+                    return false;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs:335]: The way this was designed was to be that we allow DBuffer only on platforms that we specifically approved it, while it seems to me that we should have default enabled and disable DBuffer only on platforms that we know are limited in terms of power. We expected that new generations of platforms comes with more processing power.

My justification is that DBuffer is the higher quality / more complete feature set, so now let's say we have a new generation of platform so now that platform gets less quality until someone adds the platform to this switch case, which seems counterintuitive to me.

Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs:335]: The other assumption I have is that it seems the less powerful platforms are the smaller set, so this switch case grows less than the opposite.

Comment (lukaschod) [reply com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs:335]: It makes sense, but switch statement will be much bigger

**Review by phi-lira — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs [hunk 2]
```diff
@@ -312,26 +312,15 @@ private DecalTechnique GetTechnique(bool isDeferred)
...
-                return true;
-            return false;
+            if (UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.WebGL)
+                return false;
 #else
-            return SystemInfo.deviceType == DeviceType.Desktop || SystemInfo.deviceType == DeviceType.Console;
+            if (Application.platform == RuntimePlatform.WebGLPlayer)
+                return false;
 #endif
+            return GraphicsSettings.HasShaderDefine(BuiltinShaderDefine.SHADER_API_MOBILE);
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs:323]: Should't it be opposite? You have DBuffer disableds by default in mobile?

File: com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs [hunk 3]
```diff
@@ -312,26 +312,15 @@ private DecalTechnique GetTechnique(bool isDeferred)
...
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.PS5)
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.WSA)
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.Switch)
-                return true;
-            return false;
+            if (UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.WebGL)
+                return false;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs:318]: Is this a bug? I would expect SHADER_API_MOBILE to not be used in WebGL?

**Review by phi-lira — APPROVED** [review]

**Review by ernestasKupciunas — COMMENTED** [review]
It seems like Screen Space technique is not rendered at all.
I have already tested on Mobile devices, and the Editor when either iOS or Android platform is selected.
It does render in the Scene view, but not the Game view. 
When Mobile platform is elected, and Automatic technique is used, it does indeed use the Screen Space technique, but Screen Space technique seems to be broken, so nothing is rendered in the final result.
Screenshot 2022-02-15 at 12 03 35

https://user-images.githubusercontent.com/31403375/154045089-012d70c4-90fb-4d14-835d-4eccd47e6e4a.mp4

Devices under testing:
MacBook Pro (16-inch, 2021), CPU: Apple M1 Max, Memory: 64 GB, macOS: Monterey 12.1
VLNQA00268, Samsung Galaxy S10+, 10.0.0, Exynos 9 9820, Mali-G76
Samsung Galaxy Z Fold2 5G, 10, Snapdragon 865 SM8250, Adreno 650
iPhone 13Pro Max, SoC: A15, iOS: 15.0.1

**Review by ernestasKupciunas — APPROVED** [review]
After enabling Depth Textures it works as intended. 
When Automatic technique is selected, on iOS and Android devices ScreenSpace technique is used in the Editor and Builds.
Screenshot 2022-02-15 at 14 05 51

- iOS devices do not render the decals when ScreenSpace technique is selected + MSAA is enabled in the URP asset. Will report this bug later. Not related to this PR.

Other than that, it looks good.

Devices used for testing:
MacBook Pro (16-inch, 2021), CPU: Apple M1 Max, Memory: 64 GB, macOS: Monterey 12.1
iPhone 13Pro Max, SoC: A15, iOS: 15.0.1
iPhone XS, SoC: A12, iOS: 14.3
VLNQA00268, Samsung Galaxy S10+, 10.0.0, Exynos 9 9820, Mali-G76
OnePlus Nord, 10, Snapdragon 765G SM7250-AB, Adreno 620
Samsung Galaxy Z Fold2 5G, 10, Snapdragon 865 SM8250, Adreno 650

Used Metal Graphics API for iOS, and Vulkan and Gles3 for Android

**Pull request merged**